setup/src/Magento/Setup/Module/Di/Code/Reader/FileScanner.php

```diff
@@ -207,7 +206,7 @@ protected function scan()
                     goto SCANNER_NAMESPACE_CONTINUE;
                 }
 
-                if ($this->tokenType === T_NS_SEPARATOR || $this->tokenType === T_STRING) {
+                if (\in_array($this->tokenType, $namespaceContentTokenTypes, true)) {
```

Comment:
Could you replace it with a map and looking for a key in this map? 
It works way faster, especially performance is critical for DI file scanner

Reply:
thanks for the tip, done